modules/Bio/EnsEMBL/Variation/VCFCollection.pm

```diff
@@ -486,59 +487,79 @@ sub get_all_VariationFeatures_by_Slice {
+      my @description = split(/Format:/,$desc);
+      my @info_format = split('\|', $description[1]);
+      foreach my $vf (@vfs) { 
+        if ( $vcf_info_field ){
+          my $info = $vf->{vcf_record}->get_info;
+          my @vcf_variants = $info->{$vcf_info_field};
+          foreach my $vcf_variant ( @vcf_variants ) {
+            if ( defined($vcf_variant) ){
+              my @transcript_vars = split(',',$vcf_variant);
+              foreach my $transcript_var ( @transcript_vars ){
+                my @transcript_split = split('\|',$transcript_var);
+                my %transcript_map = zip @info_format, @transcript_split;
+                next if not defined($transcript_map{'Feature_type'});
+                if ( $transcript_map{'Feature_type'} eq 'Transcript') {
+                  my $tv = Bio::EnsEMBL::Variation::TranscriptVariation->new_fast({
+                    variation_feature  => $vf,
+                    _feature_stable_id => $transcript_map{'Feature'}
+                  });  
+                  #Add variation Feature seq from transcript map
+                  my $tva = Bio::EnsEMBL::Variation::TranscriptVariationAllele->new_fast({
+                    variation_feature_seq => $transcript_split[0],
+                    is_reference               => 0
+                  });
+                  my @cons_list = split('&',$transcript_map{'Consequence'});
+                  foreach my $con (@cons_list){
+                    if ( exists $OVERLAP_CONSEQUENCES{$con} ) {
+                      my $new_cons = $OVERLAP_CONSEQUENCES{$con};
+                      $tva->add_OverlapConsequence($new_cons);
+                    }
+                    else{
+                      print("The consequence is not available:",$con,"\n");
+                    }
                   }
+                  $tv->add_TranscriptVariationAllele($tva);
+                  $vf->add_TranscriptVariation($tv);
                 }
-                $tv->add_TranscriptVariationAllele($tva);
-                $vf->add_TranscriptVariation($tv);
               }
             }
           }
         }
+        else{ 
+          $vf->{intergenic_variation} = Bio::EnsEMBL::Variation::IntergenicVariation->new(
+            -variation_feature  => $vf,
+            -no_ref_check       => 1,
+          );
+          weaken($vf->{intergenic_variation}->{base_variation_feature});
+        }
+        $vf->_finish_annotation();
       }
-      else{ 
-        $vf->{intergenic_variation} = Bio::EnsEMBL::Variation::IntergenicVariation->new(
-          -variation_feature  => $vf,
-          -no_ref_check       => 1,
-        );
-        weaken($vf->{intergenic_variation}->{base_variation_feature});
+    } elsif ($vcf_info_field eq 'CLNSIG'){
+      foreach my $vf (@vfs) {
+        my $info = $vf->{vcf_record}->get_info;
+        my $vcf_variants = $info->{$vcf_info_field};
+        my @clnsig_vars = split('[\|,/]',$vcf_variants);
```

Comment:
As there are clinsig terms with commas, you can deal with the two specific terms here.
```
my $vcf_variants = $info->{$vcf_info_field};
# replace commas for specific examples
$vcf_variants =~ s/Pathogenic,_low_penetrance/Pathogenic_low_penetrance/;
$vcf_variants =~ s/Likely_pathogenic,_low_penetrance/Likely_pathogenic_low_penetrance/;
my @clnsig_vars = split('[\|,/]',$vcf_variants);
```